frontend/src/pages/RegistrationPage.tsx

```diff
@@ -93,13 +97,26 @@ export default function RegistrationPage() {
             )}
           />
           <FormError message={errors.password_confirmation?.message} />
-          <input
-            className='btn btn-info mt-5 w-1/2 mx-auto'
-            type='submit'
-            value='Register'
-          />
+          <button className='btn btn-info mt-5 w-1/2 mx-auto' type='submit'>
+            {props.loading ? (
+              <div className='flex items-center justify-center'>
+                <div className='w-4 h-4 border-b-2 border-white-900 rounded-full animate-spin mr-5'></div>
+                Register
+              </div>
```

Comment:
Please use `<span/>` instead of `<div/>` according to w3c it is not valid.
check it in [w3c validator](https://validator.w3.org/nu/#textarea)